modules/apigee_edge_apidocs/src/Entity/ApiDoc.php

```diff
@@ -122,16 +139,48 @@ public function setCreatedTime(int $timestamp) : ApiDocInterface {
   /**
    * {@inheritdoc}
    */
-  public function isPublished() : bool {
-    return (bool) $this->getEntityKey('status');
+  public function getRevisionUser() {
+    return $this->get('revision_user')->entity;
   }
 
   /**
    * {@inheritdoc}
    */
-  public function setPublished(bool $published) : ApiDocInterface {
-    $this->set('status', $published);
-    return $this;
+  public function preSave(EntityStorageInterface $storage) {
+    parent::preSave($storage);
+
+    // If no revision author has been set explicitly, make the current user
+    // the revision author.
+    if (!$this->getRevisionUser()) {
+      $this->setRevisionUserId(\Drupal::currentUser()->id());
+    }
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function preSaveRevision(EntityStorageInterface $storage, \stdClass $record) {
+    parent::preSaveRevision($storage, $record);
+
+    if (!$this->isNewRevision() && isset($this->original) && empty($record->revision_log_message)) {
+      // If we are updating an existing entity without adding a new revision, we
+      // need to make sure $entity->revision_log is reset whenever it is empty.
+      // Therefore, this code allows us to avoid clobbering an existing log
+      // entry with an empty one.
+      $record->revision_log_message = $this->original->revision_log_message->value;
+    }
+  }
+
+  /**
+   * Gets whether a new revision should be created by default.
+   *
+   * @return bool
+   *   TRUE if a new revision should be created by default.
+   */
+  public static function shouldCreateNewRevision() {
```

Comment:
Is it really necessary to be static? `@see \Drupal\node\Entity\NodeType::shouldCreateNewRevision()`

Reply:
Yes, that works, I have changed the static method and updated references.